src/test/debuginfo/repr-u128.rs

```diff
@@ -0,0 +1,35 @@
+// ignore-windows
+// min-system-llvm-version 8.0
+
+// compile-flags: -g -C no-prepopulate-passes
+
+// === GDB TESTS ===================================================================================
+
+// gdb-command: run
+
+// gdb-command:print vals
+// gdbr-check:$1 = (<error reading variable>, <error reading variable>)
```

Comment:
We no longer catch `Options` after rebasing, is this okay? @eddyb

Reply:
That doesn't make sense, what broke this?

Reply:
cc @michaelwoerister @alexcrichton @tmandry

Reply:
Is https://github.com/rust-lang/rust/pull/59897#issuecomment-489310528 related?

Reply:
No, that's just affecting miri, and also you're not using generators at all here.

Reply:
If this is regressing something, it's likely because something changed. The only thing that really looks like it's changing is that we're always telling LLVM that there's a 128-bit discriminant. My guess is that LLVM and/or GDB isn't really prepared for that and there's some pessimization that happens somewhere. This can probably be fixed by always telling LLVM about a 64-bit discriminant when it fits and 128-bit otherwise.

Or it could be something completely different. I don't really know why I'm cc'd here and I don't really know much about this test, this PR, or debugging and descriminants being passed.

Reply:
This test worked before a recent rebase, according to @JohnTitor - see https://github.com/rust-lang/rust/pull/59520/commits/ea7af710cfdf8f1d4db6b3b702c59278a589c5d3 for the post-rebase diff.

However, I don't know how when or how we ever run debuginfo tests anymore, so I can't be sure of anything.

Reply:
I checked to see if #59897 was causing this but no, I still get the `<error reading variable>` before that PR. Could also have been #59651 but I doubt it.

Reply:
Hmm, it seems that we need more investigation.

@eddyb How we try to bisect this PR?

Reply:
I'm not sure, can you squash all the changes into one commit?
That would make it easy, since you can just `git cherry-pick the-hash-of-that-commit` on top of any older `master` and build that (or even use `git bisect` but I never remember how to).
That's assuming this test can even be tested, that is.